samples/net/mqtt_publisher/CMakeLists.txt

```diff
@@ -1,4 +1,15 @@
 cmake_minimum_required(VERSION 3.13.1)
+macro(set_conf_file)
```

Comment:
In the short-term, we might want to support a pattern where both board-specifc and general .conf files are automatically picked up, instead of either board or general configuration as it is today.

I the long-term, we might want to either extend the fragment language to allow if-statements and \#include statements, or resolve #8181 and then try to drastically cut down on the amount of Kconfig fragment key-values required.

Reply:
@SebastianBoe, thanks for reply, and glad to hear that's on radar. I assume that immediate-term, doing what this patch does here is the right way.